testng-core/src/main/java/org/testng/TestRunner.java

```diff
@@ -662,9 +663,10 @@ private static Comparator<ITestNGMethod> newComparator(boolean needPrioritySort)
     return needPrioritySort ? new TestMethodComparator() : null;
   }
 
-  private boolean sortOnPriority(ITestNGMethod[] interceptedOrder) {
-    return m_methodInterceptors.size() > 1
-        || Arrays.stream(interceptedOrder).anyMatch(m -> m.getPriority() != 0);
+  private Pair<Boolean, Boolean> sortOnPriority(ITestNGMethod[] interceptedOrder) {
+    return new Pair<>(
```

Comment:
Using pair here makes too much noise.
As it is only 2 lines, used only once, just remove the method and use appropriate var names.

Reply:
Fixed. Please check now